### PR TITLE
Migrated Batch 6 of Thermal Conductivity

### PR DESCRIPTION
**Pull request by qur786**
sir please checkout the data of Nitrogen TriFluoride , Its all  data is missing

### CODE REVIEW

File: src/data/thermal-conductivity-constants.ts [hunk 1]
```diff
@@ -2774,6 +2774,676 @@ export const DIPPRThermalConductivityDictionary: Record<
...
+    minimumTemperature: 353.43,
+    maximumTemperature: 646.97,
+    thermalConductivityAtMinimumTemperature: 0.1354,
+    thermalConductivityAtMaximumTemperature: 0.1059,
+  },
+
+  Neon: {
+    name: 'Neon',
+    formula: 'Ne',
+    CAS: '09-01-7440',
```
Comment (saikat-samanta) [inline src/data/thermal-conductivity-constants.ts:3240]: 7440-01-9

File: src/data/thermal-conductivity-constants.ts [hunk 2]
```diff
@@ -2774,6 +2774,676 @@ export const DIPPRThermalConductivityDictionary: Record<
...
+    minimumTemperature: 253.05,
+    maximumTemperature: 492.95,
+    thermalConductivityAtMinimumTemperature: 0.1484,
+    thermalConductivityAtMaximumTemperature: 0.0972,
+  },
+
+  '1Nonyne': {
+    name: '1-Nonyne',
+    formula: 'C9H16',
+    CAS: '03-09-3452',
```
Comment (saikat-samanta) [inline src/data/thermal-conductivity-constants.ts:3438]: 3452-09-3

**Review by saikat-samanta — APPROVED** [review]
Rest looks good 👍

File: src/data/thermal-conductivity-constants.ts [hunk 3]
```diff
@@ -2774,6 +2774,676 @@ export const DIPPRThermalConductivityDictionary: Record<
...
+    name: 'Nitrogen trifluoride',
+    formula: 'F3N',
+    CAS: '7783-54-2',
+    molecularWeight: 71.00191,
+    C1: NaN,
+    C2: NaN,
+    minimumTemperature: NaN,
+    maximumTemperature: NaN,
+    thermalConductivityAtMinimumTemperature: NaN,
+    thermalConductivityAtMaximumTemperature: NaN,
```
Comment (maneetgoyal) [inline src/data/thermal-conductivity-constants.ts:3288]:
```suggestion
```

The interface should be adopted accordingly.

Comment (maneetgoyal) [reply src/data/thermal-conductivity-constants.ts:3288]: - [ ] Also add a test case (Jest) for this particular compound.

File: src/data/thermal-conductivity-constants.ts [hunk 4]
```diff
@@ -2774,6 +2774,676 @@ export const DIPPRThermalConductivityDictionary: Record<
...
+    thermalConductivityAtMaximumTemperature: 0.176,
+  },
+
+  NitrousOxide: {
+    name: 'Nitrous oxide',
+    formula: 'N2O',
+    CAS: '10024-97-2',
+    molecularWeight: 44.0128,
+    C1: 0.10112,
+    C2: NaN,
```
Comment (maneetgoyal) [inline src/data/thermal-conductivity-constants.ts:3310]:
```suggestion
```

Update the interface as needed.

Comment (maneetgoyal) [reply src/data/thermal-conductivity-constants.ts:3310]: - [ ] Also add a test case for this compound.

**Review by maneetgoyal — COMMENTED** [review]
Some refactoring needed. Some test cases should also be added.

File: src/data/thermal-conductivity-constants.ts (hunk 4 above)
Comment (qur786) [reply src/data/thermal-conductivity-constants.ts:3310]: sir , I dont think that the test case for this compound

Comment (qur786) [reply src/data/thermal-conductivity-constants.ts:3310]: is needed

File: src/data/thermal-conductivity-constants.ts [hunk 5]
```diff
@@ -2784,11 +3447,13 @@ export function calculateDIPPRThermalConductivity(
   let thermalConductivity = NaN;
   if (
     thermalConductivityProps !== undefined &&
-    temperature >= thermalConductivityProps.minimumTemperature &&
-    temperature <= thermalConductivityProps.maximumTemperature
+    temperature >= (thermalConductivityProps.minimumTemperature as number) &&
+    temperature <= (thermalConductivityProps.maximumTemperature as number)
```
Comment (maneetgoyal) [inline src/data/thermal-conductivity-constants.ts:3451]: `as assertion` should not be used. Instead the function should be modified to handle `undefined` value.

**Review by maneetgoyal — CHANGES_REQUESTED** [review]

File: src/data/thermal-conductivity-constants.ts (hunk 5 above)
Comment (qur786) [reply src/data/thermal-conductivity-constants.ts:3451]: @maneetgoyal but sir , I had to make thermal conductivity at Tmax and Tmin optional . So I have to use number.isfinit() function with as assertion

Comment (sumitnegi7) [reply src/data/thermal-conductivity-constants.ts:3451]:
```js
export function calculateDIPPRThermalConductivity(
  compound: string,
  temperature: number
): number {
  const thermalConductivityProps: DIPPRThermalConductivityProps | undefined =
    DIPPRThermalConductivityDictionary[compound];
  let thermalConductivity = NaN;

  if (
    thermalConductivityProps !== undefined &&
    thermalConductivityProps.minimumTemperature !== undefined &&
    thermalConductivityProps.maximumTemperature !== undefined &&
    temperature >= thermalConductivityProps.minimumTemperature &&
    temperature <= thermalConductivityProps.maximumTemperature
  ) {
    thermalConductivity =
      (Number.isFinite(thermalConductivityProps.C1)
        ? (thermalConductivityProps.C1 as number)
        : 0) +
      (Number.isFinite(thermalConductivityProps.C2)
        ? (thermalConductivityProps.C2 as number) * temperature
        : 0) +
      (Number.isFinite(thermalConductivityProps.C3)
        ? Math.pow(temperature, 2) * (thermalConductivityProps.C3 as number)
        : 0) +
      (Number.isFinite(thermalConductivityProps.C4)
        ? Math.pow(temperature, 3) * (thermalConductivityProps.C4 as number)
        : 0) +
      (Number.isFinite(thermalConductivityProps.C5)
        ? Math.pow(temperature, 4) * (thermalConductivityProps.C5 as number)
        : 0);
  }

  return thermalConductivity;
}

```

This should work i think incase the `minimumTemperature ` or`maximumTemperature ` are undefined we return `NaN`.

Comment (maneetgoyal) [reply src/data/thermal-conductivity-constants.ts]: >does this suggestion seems okay to you?

Yeah it sounds good. The current implementation uses `typeof` check which is a bit more robust though.

**Review by maneetgoyal — APPROVED** [review]
👍

File: src/data/thermal-conductivity-constants.ts (hunk 4 above)
Comment (maneetgoyal) [reply src/data/thermal-conductivity-constants.ts]: Test case is preferred here

File: src/data/thermal-conductivity-constants.ts (hunk 3 above)
Comment (maneetgoyal) [reply src/data/thermal-conductivity-constants.ts]: Was the test case added?

File: src/data/thermal-conductivity-constants.ts (hunk 4 above)
Comment (qur786) [reply src/data/thermal-conductivity-constants.ts]: ok sir I will add the test case for this compound

File: src/data/thermal-conductivity-constants.ts (hunk 3 above)
Comment (qur786) [reply src/data/thermal-conductivity-constants.ts]: yeah , I have added it ,

**Review by sumitnegi7 — APPROVED** [review]
LGTM :+1:

**Review by maneetgoyal — APPROVED** [review]
👍

**Pull request merged**